plugins/com.google.cloud.tools.eclipse.appengine.libraries.test/src/com/google/cloud/tools/eclipse/appengine/libraries/PomTest.java

```diff
@@ -0,0 +1,194 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.eclipse.appengine.libraries;
+
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.parsers.ParserConfigurationException;
+import javax.xml.soap.Node;
+
+import org.eclipse.core.resources.IFile;
+import org.eclipse.core.resources.IProject;
+import org.eclipse.core.runtime.CoreException;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Rule;
+import org.junit.Test;
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+import org.w3c.dom.NodeList;
+import org.xml.sax.SAXException;
+
+import com.google.cloud.tools.eclipse.appengine.libraries.model.Library;
+import com.google.cloud.tools.eclipse.appengine.libraries.model.LibraryFile;
+import com.google.cloud.tools.eclipse.appengine.libraries.model.MavenCoordinates;
+import com.google.cloud.tools.eclipse.test.util.project.TestProjectCreator;
+
+public class PomTest {
+  
+  @Rule public final TestProjectCreator projectCreator = new TestProjectCreator();
+
+  // todo we're doing enough of this we should import or write some utilities
+  private static final DocumentBuilderFactory factory = DocumentBuilderFactory.newInstance();
+  
+  @BeforeClass 
+  public static void configureParser() {
+    factory.setNamespaceAware(true);
+  }
+  
+  private Pom pom;
+  private IFile pomFile;
+  
+  @Before
+  public void setUp() throws SAXException, IOException, CoreException {
+    IProject project = projectCreator.getProject();
+    pomFile = project.getFile("pom.xml");
+    InputStream in = new FileInputStream(
+       Paths.get("testdata/testpom.xml").toAbsolutePath().toFile());
+    pomFile.create(in, IFile.FORCE, null);
+   
+    pom = Pom.parse(pomFile);
+  }
+  
+  @Test
+  public void testAddDependencies() 
+      throws CoreException, ParserConfigurationException, IOException, SAXException {
+    
+    MavenCoordinates coordinates0 = new MavenCoordinates("com.example.group0", "artifact0");
+    coordinates0.setVersion("1.2.3");
+    LibraryFile file0 = new LibraryFile(coordinates0);
+    List<LibraryFile> list0 = new ArrayList<>();
+    list0.add(file0);
+    
+    List<LibraryFile> list1 = new ArrayList<>();
```

Comment:
It may make sense to extract this and the following 2 lines into a helper method and use it to create all `Library`s/`List<LibraryFile>`s.